VIDEO_4_SCRIPT.md

```diff
@@ -0,0 +1,50 @@
+# Video 4: AsyncAPI and OpenAPI
+
+Hello again, and welcome to a new video about AsyncAPI. In this video, we're going to talk about:
+
+AsyncAPI and OpenAPI
+
+•	What is OpenAPI?
+•	Similarities between AsyncAPI and OpenAPI
+•	Differences between them
+•	Example of each initiative
+
+Here we go! 
+
+## What is OpenAPI?
+
+OpenAPI is a standard for describing application programming interfaces (APIs). This specification defines an open, vendor-independent description format for API services. In particular, OpenAPI can describe, develop, test, and document REST-compliant APIs.
+
+The current OpenAPI specification grew out of the predecessor project Swagger. The development company SmartBear subjected the existing Swagger specification to an open license, handing over both maintenance and development of the initiative to OpenAPI. In addition to SmartBear, members of the OpenAPI initiative include industry giants such as Google, IBM, or Microsoft. The Linux Foundation also supports this project. 
+
+## Similarities between AsyncAPI and OpenAPI
+
+Its ultimate goal is to work with EDA architectures as simply as with Rest APIs. This specification for the asynchronous world is analogous to OpenAPI in the synchronous world. Both initiatives are based on the same principles:
+
+- Their use is intended for a contract-first philosophy.
+
+- They are implementation agnostic.
+
+- Both use the same syntax. 
+
+- Like AsyncAPI, OpenAPI also offers a set of tools around it.
+
+- They protect information consumption using different security schemes such as user and password, API keys, certificates, and of course, the most recommended ones such as OAuth or OIDC flows.
+
+- Both specifications are extensible by including new properties with the prefix x-.
+
+## Differences between them
+
+As we have seen, the two initiatives are very similar but cover very different fields. REST APIs have a specific function that requires their methods and documentation approaches. REST APIs have, among others, a great solution in the form of OpenAPI, a set of standardized documentation methods for enabling tools and interactivity.
+
+However, message-based APIs are very different and need solutions covering the caveats and considerations intrinsic to message-based architectures. AsyncAPI's tagline says it best: "REST APIs have OpenAPI. Messaging has AsyncAPI."
+
+It's essential to understand that while the two solutions have a lot in common, they are also wildly divergent. Both OpenAPI and AsyncAPI allow for more significant cooperation and collaboration between different documented APIs. Still, OpenAPI is a solution for REST, and AsyncAPI is a solution for message-based architectures. Each is aimed at a different domain.
+
+## Example of each initiative
+
+(How about creating here a practical and simple example of each initiative? I think it would make the video more complete.)
```

Comment:
## AsyncAPI and OpenAPI comparison 

As you can see, you'll find lots of similarities between OpenAPI and AsyncAPI. In the world of event-driven architectures, you have more than one protocol, and therefore, some things are different. Let's check out the following comparison chart, inspired by [Darrel Miller's blog post](https://www.openapis.org/news/blogs/2016/10/tdc-structural-improvements-explaining-30-spec-part-2):

<img width="450" alt="Screen Shot 2022-04-01 at 4 17 06 PM" src="https://user-images.githubusercontent.com/19964402/161353528-23fd7a02-7917-4e1b-856d-c66b8456f373.png">

Reply:
I like the idea. Like this, the differences and similarities of each initiative are very clear. Do you think we should describe verbally what appears in the image?

Reply:
I can see how that could be useful! I vote yes... do you want me to help add this myself? (I am not sure I would get it 100% technically correct either, I will have to ask for another techy review.)

@Barbanio

Reply:
Ok, let me add something and then tag Lukasz when I'm done because I'm not sure I will get it right either 🤣 

I will tag you again here once I add my first attempt 😄

Reply:
## AsyncAPI and OpenAPI comparison (draft 2)

As you can see, you'll find lots of similarities between OpenAPI and AsyncAPI!

<img width="300" alt="Screen Shot 2022-04-01 at 4 23 25 PM" src="https://user-images.githubusercontent.com/19964402/161353808-430cb73e-e2a3-4913-ad5e-6cccdeeaa997.png">

In event-driven architectures, you have more than one protocol, and therefore, some things are different. Let's check out the following comparison chart, partially inspired by [Darrel Miller's blog post](https://www.openapis.org/news/blogs/2016/10/tdc-structural-improvements-explaining-30-spec-part-2):

<img width="300" alt="Screen Shot 2022-04-01 at 4 17 06 PM" src="https://user-images.githubusercontent.com/19964402/161353528-23fd7a02-7917-4e1b-856d-c66b8456f373.png">

As we look at these diagrams, we see how AsyncAPI uses `servers` for configuring their `hosts` and `security`, while OpenAPI mentions only **hosts** and **security** handled separately. Moving on, AsyncAPI has `channels` with `messages` that include `headers` and `payload` information. OpenAPI, however, has **paths** for making **REST API** calls. Lastly, notice how both AsyncAPI and OpenAPI have **`components`**. In fact, both AsyncAPI and OpenAPI have a **`Parameters`** component in common, but the rest of their components are completely different.


___

Ok, how about something like that, @Barbanio? 🤔 💭  

We will still need someone like @derberg or @smoya to review my proposal here and confirm if it's technically acurate. 😄 👍🏻

Reply:
<img width="781" alt="Screen Shot 2022-04-06 at 5 55 18 PM" src="https://user-images.githubusercontent.com/19964402/162098862-af7e12f4-c297-46d4-9035-b7e97c2e4dfc.png">

@derberg / @smoya / etc.. the section enclosed by the blue 🟦  is what needs technical review